controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -259,11 +277,24 @@ func enforcePath(authoritative, twoWay map[string]interface{}, path contract.Pat
 
 	default:
 		// If in the middle of a path, go into the nested map,
-		nestedSimpleMap, ok := authoritative[path[0]].(map[string]interface{})
+		var nestedSimpleMap map[string]interface{}
+		switch v, ok := authoritative[path[0]]; {
+		case ok && v == nil:
+			// If the nested map is nil, it means that the authoritative patch is trying to delete a parent object
+			// in the middle of the enforced path.
+
+			// If the parent object has been intentionally deleted (the corresponding parent object value in the modified object is null),
+			// then we should enforce the deletion of the parent object including everything below it.
+			if _, ok := modified[path[0]]; ok && v == nil {
+				twoWay[path[0]] = nil
+				return
+			}
 
-		// If the value is not a map (is a value or a list), return (not a full match).
-		if !ok {
-			return
+			// Otherwise, we continue processing the enforced path, thus deleting only what
+			// is explicitly enforced.
+			nestedSimpleMap = map[string]interface{}{path[1]: nil}
+		default:
+			nestedSimpleMap, _ = v.(map[string]interface{})
```

Comment:
Shouldn't this throw a panic if `ok == false && v == nil` ?


```go
	v := map[string]interface{}{
		"abc": nil,
	}
	a := v["abc"].(map[string]interface{}) // panic
```

Reply:
v being nil is handled by the first case in the switch statement; everything else should always be a map[string]interface{} given how Unmarshal works, but I have added a safeguard to avoid panics down the line.

Reply:
My thought was that the first case only handles `ok && v==nil` but if `ok == false && v == nil` is impossible that's fine too :)